### PR TITLE
Convert the API to return a Promise

### PR DESCRIPTION
**Pull request by christophehurpeau**
This PR add support for Promise return, and keeping callback work.

### CODE REVIEW

**Comment by SamVerschueren** [comment]
Drop callback.

File: readme.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ githubUsername('sindresorhus@gmail.com', function (err, username) {
 
 ## API
 
-### githubUsername(email, [token], callback)
+### githubUsername(email, [token], [callback]): Promise<username>
```
Comment (sindresorhus) [inline readme.md]: The correct notation would be `Promise<string>`, but I would just write this below instead:

> Returns a `Promise` for the username.

**Comment by sindresorhus** [comment]
Thanks for the PR. What Sam said. I'm converting all my modules to Promise API only.

**Comment by sindresorhus** [comment]
Awesome! Thanks :)

**Pull request merged**